site/src/contexts/ProxyContext.tsx

```diff
@@ -6,25 +6,60 @@ import {
   createContext,
   FC,
   PropsWithChildren,
+  useCallback,
   useContext,
+  useEffect,
   useState,
 } from "react"
 import { ProxyLatencyReport, useProxyLatency } from "./useProxyLatency"
 
-interface ProxyContextValue {
+export interface ProxyContextValue {
+  // proxy is **always** the workspace proxy that should be used.
+  // The 'proxy.selectedProxy' field is the proxy being used and comes from either:
+  //   1. The user manually selected this proxy. (saved to local storage)
+  //   2. The default proxy auto selected because:
+  //    a. The user has not selected a proxy.
+  //    b. The user's selected proxy is not in the list of proxies.
+  //    c. The user's selected proxy is not healthy.
+  //   3. undefined if there are no proxies.
+  //
+  // The values 'proxy.preferredPathAppURL' and 'proxy.preferredWildcardHostname' can
+  // always be used even if 'proxy.selectedProxy' is undefined. These values are sourced from
+  // the 'selectedProxy', but default to relative paths if the 'selectedProxy' is undefined.
   proxy: PreferredProxy
+
+  // userProxy is always the proxy the user has selected. This value comes from local storage.
+  // The value `proxy` should always be used instead of `userProxy`. `userProxy` is only exposed
+  // so the caller can determine if the proxy being used is the user's selected proxy, or if it
+  // was auto selected based on some other criteria.
+  //
+  // if(proxy.selectedProxy.id === userProxy.id) { /* user selected proxy */ }
+  // else { /* proxy was auto selected */ }
```

Comment:
```suggestion
```

Reply:
That is actually intentional!

I was leaving this comment to tell the reader this is the conditional you can use to determine if the proxy being used to generate links is the same proxy that the user requested.

This information is a bit excessive, but defers the UX to the caller of the context.

The situation where this is helpful is:

1. Regions `North America`, `South America`, and `Europe` exist.
2. South American user has `South America` selected.
3. `South America` proxy is unhealthy, by latency `North America` is selected.
4. The user now sees their proxy is `North America` and might be confused as they never selected that.

With this information exposed, the UX can display `South America` as some disabled proxy, or something else to tell the user "We know you selected `South America`, and if it comes back online, you will be switched back. But at present, it is unavailable, so we defaulted to a new region for you".

Reply:
Got it! Thanks for the explanation!